mvpa2/algorithms/hyperalignment.py

```diff
@@ -493,3 +506,21 @@ def _level3(self, datasets):
                 residuals[0, i] = np.linalg.norm(data_mapped - self.commonspace)
 
         return mappers
+
+    def _map_and_mean(self, datasets, mappers):
+        params = self.params
+        data_mapped = [[] for ds in datasets]
+        for i, (m, ds_new) in enumerate(zip(mappers, datasets)):
+            if  __debug__:
+                debug('HPAL_', "Mapping for SVD: ds #%i" % i)
+            ds_ = m.forward(ds_new.samples)
+            zscore(ds_, chunks_attr=None)
+            data_mapped[i] = ds_
+        dss_mean = params.combiner2(data_mapped)
+        return dss_mean
+
+    def _reduce_svd(self, dss_mean):
+        zscore(dss_mean, chunks_attr=None)
+        svd = SVDMapper(force_train=True)
+        svd.train(dss_mean)
+        return svd
```

Comment:
note that `self` is not used so you could make this a `@staticmethod` or just an independent function helper